src/cpp/database/database_queue.cpp

```diff
@@ -558,39 +620,51 @@ void DatabaseDataQueue::process_sample()
             catch (const eprosima::statistics_backend::Exception& e)
             {
                 logWarning(BACKEND_DATABASE_QUEUE,
-                        "Error processing SAMPLE_DATAS event. Data was not added to the statistics collection.");
+                        "Error processing SAMPLE_DATAS event. Data was not added to the statistics collection: "
+                        + std::string(e.what()));
             }
+            break;
         }
-        break;
         case StatisticsEventKind::PHYSICAL_DATA:
         {
             StatisticsPhysicalData item = front().second->physical_data();
 
             // Take the ID of the Participant from its GUID
             std::string participant_guid = deserialize_guid(item.participant_guid());
-            auto participants = database_->get_entities_by_guid(EntityKind::PARTICIPANT, participant_guid);
-            if (participants.empty())
-            {
-                throw BadParameter("No participant with GUID " + participant_guid + " exists");
-            }
-            EntityId participant_id = participants.front().second;
+            auto participants = database_->get_entity_by_guid(EntityKind::PARTICIPANT, participant_guid);
+            EntityId participant_id = participants.second;
 
             // Parse the process name and PID
+            std::string process_name;
+            std::string process_pid;
             size_t separator_pos = item.process().find_last_of(':');
             if (separator_pos == std::string::npos)
             {
-                throw Error("Process name " + item.process() + " does not follow the [command]:[PID] pattern");
+                logError(BACKEND_DATABASE_QUEUE,
+                        "Process name " + item.process() + " does not follow the [command]:[PID] pattern");
+                process_name = item.process();
+            }
+            else
+            {
+                process_name = item.process().substr(0, separator_pos);
+                process_pid = item.process().substr(separator_pos + 1);
             }
-            std::string process_name = item.process().substr(0, separator_pos);
-            std::string process_pid = item.process().substr(separator_pos + 1);
 
             // Check the existence of the host
             std::shared_ptr<Host> host;
```

Comment:
Codecov is saying that changes would remove coverage from this line... I do not understand why but if you can have a look it would be nice.

Reply:
It makes no sense that the testing is not going through this specific line but does cover the surrounding ones. My only guess is some compiler optimization, as the `host` variable is declared here, but overridden in both branches of the `if` below.